File: app/src/main/java/com/mamp/qrscanner/db/DbContract.java
package com.mamp.qrscanner.db;

import android.provider.BaseColumns;

public class DbContract {

    private DbContract() {}

    public static final class QrData implements BaseColumns {
        public static final String TABLE_NAME = "qr_data";
        public static final String COLUMN_QR_DATA = "qr_data";
        public static final String COLUMN_INPUT_DATE = "input_date";

        public static final String SQL_CREATE_QR_DATA =
                "CREATE TABLE " + TABLE_NAME + " (" +
                        _ID + " INTEGER PRIMARY KEY," +
                        COLUMN_QR_DATA + " TEXT," +
                        COLUMN_INPUT_DATE + " TEXT)";

        public static final String SQL_DROP_QR_DATA =
                "DROP TABLE IF EXISTS " + TABLE_NAME;
    }


}

File: app/src/main/java/com/mamp/qrscanner/db/DbContract.kt
package com.mamp.qrscanner.db

import android.provider.BaseColumns

object DbContract {
    object QrData : BaseColumns {
        const val TABLE_NAME = "qr_data"
        const val COLUMN_QR_DATA = "qr_data"
        const val COLUMN_INPUT_DATE = "input_date"

        // 테이블 생성 SQL 문
        const val SQL_CREATE_QR_DATA =
            "CREATE TABLE $TABLE_NAME (" +
                    "${BaseColumns._ID} INTEGER PRIMARY KEY," +
                    "$COLUMN_QR_DATA TEXT," +
                    "$COLUMN_INPUT_DATE TEXT)"

        // 테이블 삭제 SQL 문
        const val SQL_DROP_QR_DATA = "DROP TABLE IF EXISTS $TABLE_NAME"
    }
}
